packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -80,15 +87,31 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     this.multitouchDragStrategy = MultitouchDragStrategy.latestPointer,
     this.velocityTrackerBuilder = _defaultBuilder,
     this.onlyAcceptDragOnThreshold = false,
+    this.cancelWhenOutsideBoundary = false,
+    this.createDragBoundary,
     super.supportedDevices,
     AllowedButtonsFilter? allowedButtonsFilter,
-  }) : super(allowedButtonsFilter: allowedButtonsFilter ?? _defaultButtonAcceptBehavior);
+  }) : assert(
+          !cancelWhenOutsideBoundary || createDragBoundary != null
+       ), super(allowedButtonsFilter: allowedButtonsFilter ?? _defaultButtonAcceptBehavior);
 
   static VelocityTracker _defaultBuilder(PointerEvent event) => VelocityTracker.withKind(event.kind);
 
   // Accept the input if, and only if, [kPrimaryButton] is pressed.
   static bool _defaultButtonAcceptBehavior(int buttons) => buttons == kPrimaryButton;
 
+  /// Called when the drag gesture starts. The returned [DragBoundary] will
+  /// be used to define the boundary for this drag gesture.
+  ///
+  /// See also:
+  /// * [DragBoundary], which defines the boundary of the drag gesture.
+  /// * [CreateDragBoundary], which is a callback that creates a [DragBoundary].
+  CreateDragBoundary? createDragBoundary;
+
+  /// Whether to cancel the gesture when it moves out of the boundary
+  /// specified by [createDragBoundary], defaults to false.
+  bool cancelWhenOutsideBoundary;
```

Comment:
Can you provide an example on how to use this API to fix the issue described in https://github.com/flutter/flutter/issues/146749 .